geo: allow case insensitive SRID= at the start of EWKT

otan opened the pull request:
PostGIS allows any combination of `srid` at the beginning of the string.
Follow the same allowance by having a case insensitive HasPrefix.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/48551)
<!-- Reviewable:end -->

sumeerbhola reviewed (changes requested):
> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/48551)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan)

___
*[pkg/geo/parse.go, line 194 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/48551#-M6oLuKlB5RFFDcJGDsv:-M6oLuKlB5RFFDcJGDsw:b-6un2a9) ([raw file](https://github.com/cockroachdb/cockroach/blob/6f076099bb72e72b2d6338ff004106f83c2ad125/pkg/geo/parse.go#L194)):*
> ```Go
> 	}
> 	for i := 0; i < len(prefix); i++ {
> 		if unicode.ToLower(rune(str[i])) != unicode.ToLower(rune(prefix[i])) {
> ```

It is not clear to me why this is correct in the general utf-8 case (though fine for SRID): my understanding is that the string index returns a byte, not a rune. Can you fix that or add a comment that this works only for a prefix that is ascii.


<!-- Sent from Reviewable.io -->

otan commented:
___
*[pkg/geo/parse.go, line 194 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/48551#-M6oLuKlB5RFFDcJGDsv:-M6onJqc-D5Z9ss1sxJ-:bcf5sse) ([raw file](https://github.com/cockroachdb/cockroach/blob/6f076099bb72e72b2d6338ff004106f83c2ad125/pkg/geo/parse.go#L194)):*
<details><summary><i>Previously, sumeerbhola wrote…</i></summary><blockquote>

It is not clear to me why this is correct in the general utf-8 case (though fine for SRID): my understanding is that the string index returns a byte, not a rune. Can you fix that or add a comment that this works only for a prefix that is ascii.
</blockquote></details>

I'll just cop it and write the builtin myself.


<!-- Sent from Reviewable.io -->

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 3 of 3 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/48551#-:-M6pWfV7CtIXzrqozp-0:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @otan)

___
*[pkg/geo/parse.go, line 194 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/48551#-M6oLuKlB5RFFDcJGDsv:-M6pVxTSCBtq6cq8FYh9:b-ctd50v) ([raw file](https://github.com/cockroachdb/cockroach/blob/6f076099bb72e72b2d6338ff004106f83c2ad125/pkg/geo/parse.go#L194)):*
<details><summary><i>Previously, otan (Oliver Tan) wrote…</i></summary><blockquote>

I'll just cop it and write the builtin myself.
</blockquote></details>

Can you add the following to the function level comment.
// It assumes that the prefix contains only ASCII bytes.
___
*[pkg/util/strings.go, line 34 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/48551#-M6pVKH4D0VPP-2zbYKV:-M6pVKH4D0VPP-2zbYKW:b-hj425m) ([raw file](https://github.com/cockroachdb/cockroach/blob/c760c5f4c88e772bf3eaec9b6b19ee665204cd28/pkg/util/strings.go#L34)):*
> ```Go
> }
> 
> // ToLowerSingleByte returns the the lowercase of a given single ASCII byte.
> ```

can you add 
// A non ASCII byte is returned unchanged.

and add a couple of non-ascii bytes to the test.


<!-- Sent from Reviewable.io -->

otan committed:
geo: allow case insensitive SRID= at the start of EWKT

PostGIS allows any combination of `srid` at the beginning of the string.
Follow the same allowance by having a case insensitive HasPrefix.

Release note: None

otan commented:
tftr!

bors r=sumeerbhola
